packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -124,64 +172,60 @@ class FlutterPlugin implements Plugin<Project> {
             if (!engineOut.isDirectory()) {
                 throw new GradleException('localEngineOut must point to a local engine build')
             }
-            flutterJar = Paths.get(engineOut.absolutePath, "flutter.jar").toFile()
+            baseEnginePath = Paths.get(engineOut.absolutePath)
+            flutterJar = baseEnginePath.resolve("flutter.jar").toFile()
             if (!flutterJar.isFile()) {
                 throw new GradleException('File not found: ' + flutterJar)
             }
 
             localEngine = engineOut.name
             localEngineSrcPath = engineOut.parentFile.parent
-
-            project.dependencies {
-                if (project.getConfigurations().findByName("api")) {
-                    api project.files(flutterJar)
-                } else {
-                    compile project.files(flutterJar)
-                }
-            }
+            debugFlutterJar = flutterJar
```

Comment:
I'm not sure if it's better to save all these variables to the same kind of flutter jar in this case or leave the ones that don't fit the engine build type null. Either way I think a comment here explaining why these are getting their types clobbered would be good. (/cc @dnfield in case he knows)

Reply:
My understanding is that this shouldn't change the current behavior - I added a comment to clarify. It may be worth failing the build if the engine isn't built for a particular build type. Anyways, it's clear there's still some work to do here.